### PR TITLE
Add basic ci pipeline

### PR DESCRIPTION
**Pull request by AnthonyMichaelTDM**
adds a CI pipeline and configuration file for dependabot

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+      - uses: actions/cache@v3.2.3
+        with:
+          path: |
+            ./.cargo/.build
+            ./target
+            ~/.cargo
+          key: ${{ runner.os }}-cargo-test-${{ hashFiles('**/Cargo.lock') }}
+      - uses: actions-rs/cargo@v1.0.3
+        with:
+          command: test
```
Comment (Wulf) [inline .github/workflows/CI.yml:116]: could we add another comand which runs ./test_all and checks that the exit code for [`git diff --exit-code`](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---exit-code) is 0?

**Review by Wulf — COMMENTED** [review]
Thanks for this @AnthonyMichaelTDM, I saw you've added a pipeline to create-rust-app as well. These will be super helpful!

I wanted to request if you could add a job which runs `test_all` (see comment below). If not, no worries, this is a great starting point! :raised_hands:

**Comment by AnthonyMichaelTDM** [comment]
I saw that, should be possible.

I'll try to throw something together later tonight

**Comment by AnthonyMichaelTDM** [comment]
Also, my 2 recent PR's for create-rust-app are ready to be merged as far as I'm concerned.

One has a lot of changes and the other introduces breaking changes so I'm not going to merge them myself out of respect.

**Pull request merged**